controllers/components/kserve/kserve_controller_actions.go

```diff
@@ -0,0 +1,267 @@
+package kserve
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	componentsv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/components/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/controllers/status"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster/gvk"
+	odherrors "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/errors"
+	odhtypes "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/types"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/feature"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/metadata/labels"
+)
+
+func checkPreConditions(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	k, ok := rr.Instance.(*componentsv1.Kserve)
+	if !ok {
+		return fmt.Errorf("resource instance %v is not a componentsv1.Kserve)", rr.Instance)
+	}
+
+	if k.Spec.Serving.ManagementState == operatorv1.Managed {
+		if rr.DSCI.Spec.ServiceMesh == nil || rr.DSCI.Spec.ServiceMesh.ManagementState != operatorv1.Managed {
+			s := k.GetStatus()
+			s.Phase = status.PhaseNotReady
+
+			meta.SetStatusCondition(&s.Conditions, metav1.Condition{
+				Type:               status.ConditionTypeReady,
+				Status:             metav1.ConditionFalse,
+				Reason:             status.ServiceMeshNotConfiguredReason,
+				Message:            status.ServiceMeshNotConfiguredMessage,
+				ObservedGeneration: s.ObservedGeneration,
+			})
+
+			return odherrors.NewStopError(status.ServiceMeshNotConfiguredMessage)
+		}
+
+		if found, err := cluster.OperatorExists(ctx, rr.Client, serviceMeshOperator); err != nil {
+			return odherrors.NewStopErrorW(err)
```

Comment:
I would set a proper condition here too so I would enter the else block if either `err != nil` or `!found`. In case of error, then the error can be used to set the message

Reply:
same would probably apply to the if/ten below